packages/context/src/context.ts

```diff
@@ -797,21 +799,27 @@ export class Context extends EventEmitter {
   ): ValueOrPromise<ValueType | undefined> {
     const {key, propertyPath} = BindingKey.parseKeyWithPath(keyWithPath);
 
-    optionsOrSession = asResolutionOptions(optionsOrSession);
+    const options = asResolutionOptions(optionsOrSession);
 
-    const binding = this.getBinding<ValueType>(key, optionsOrSession);
-    if (binding == null) return undefined;
-
-    const boundValue = binding.getValue(this, optionsOrSession);
-    if (propertyPath === undefined || propertyPath === '') {
-      return boundValue;
-    }
-
-    if (isPromiseLike(boundValue)) {
-      return boundValue.then(v => getDeepProperty<ValueType>(v, propertyPath));
+    const binding = this.getBinding<ValueType>(key, {optional: true});
```

Comment:
Before this change, we were passing `optionsOrSession` to `ctx.getBinding`. With your change in place, the user-provided options are discarded. Are you sure that `getBinding` is never accessing options? What if a LB consumer overwrites `getBinding` in their application to tweak the way how bindings are resolved, perhaps using the `options` flags?

IMO, we need to preserve options here. Perhaps this would work?

```ts
const binding = this.getBinding<ValueType>(key, {...options, optional: true});
```

Reply:
Not really, the signature of `getBinding` is as follows:
```ts
getBinding<ValueType>(
    key: BindingAddress<ValueType>,
    options?: {optional?: boolean},
  ): Binding<ValueType> | undefined;
```

As you see, the type of `options` is `{optional?: boolean}`.